src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -56,6 +56,80 @@
                                        ["code", "lib", "func", "reason"])
 
 
+def _encode_asn1_int(backend, x):
+    i = backend._lib.ASN1_INTEGER_new()
+    # i = backend._ffi.gc(i, backend._lib.ASN1_INTEGER_free)
+    backend._lib.ASN1_INTEGER_set(i, x)
+    return i
+
+
+def _encode_asn1_str(backend, x, n):
+    s = backend._lib.ASN1_OCTET_STRING_new()
+    # s = backend._ffi.gc(s, backend._lib.ASN1_OCTET_STRING_free)
+    backend._lib.ASN1_OCTET_STRING_set(s, x, n)
+    return s
+
+
+def _encode_name(backend, attributes):
+    resolve = {
```

Comment:
We use `OBJ_obj2txt` to grab an ASN1 object and turn it into a textual OID in backends/openssl/x509.py. We should be able to use `OBJ_txt2obj` to convert an OID back into an OBJ and then use `X509_NAME_add_entry_by_OBJ` rather than building this incomplete mapping. (`X509_NAME_add_entry_by_OBJ` will, of course, need to be added as a binding).